src/server/services/wfs3/qgswfs3handlers.cpp

```diff
@@ -0,0 +1,1178 @@
+            {
+              { "href", href( api, context.request(), QStringLiteral( "/%1/concepts" ).arg( shortName ) )  },
+              { "rel", QgsServerOgcApi::relToString( QgsServerOgcApi::Rel::item ) },
+              { "type", "text/html" },
+              { "title", "Describe " + title }
+            }
+            */
+          }
+        },
+      } );
+    }
+  }
+  json navigation = json::array();
+  const auto url { context.request()->url() };
+  navigation.push_back( {{ "title",  "Landing page" }, { "href", parentLink( url, 1 ) }} ) ;
+  write( data, context, {{ "pageTitle", linkTitle() }, { "navigation", navigation }} );
+}
+
+json Wfs3CollectionsHandler::schema( const QgsServerApiContext &context ) const
+{
+  json data;
+  const auto path { ( context.apiRootPath() + QStringLiteral( "/collections" ) ).toStdString() };
+  data[ path ] =
+  {
+    {
+      "get", {
+        { "tags", jsonTags() },
+        { "summary", summary() },
+        { "operationId", operationId() },
+        {
+          "responses", {
+            {
+              "200", {
+                { "description", description() },
+                {
+                  "content", {
+                    {
+                      "application/json", {
+                        {
+                          "schema",  {
+                            { "$ref", "#/components/schemas/content" }
+                          }
+                        }
+                      }
+                    },
+                    {
+                      "text/html", {
+                        {
+                          "schema",  {
+                            { "type", "string" }
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            },
+            defaultResponse()
+          }
+        }
+      }
+    }
+  };
+  return data;
+}
+
+Wfs3DescribeCollectionHandler::Wfs3DescribeCollectionHandler()
+{
+}
+
+void Wfs3DescribeCollectionHandler::handleRequest( const QgsServerApiContext &context ) const
+{
+  if ( ! context.project() )
+  {
+    throw QgsServerApiImproperlyConfiguredException( QStringLiteral( "Project is invalid or undefined" ) );
+  }
+  // Check collectionId
+  const auto match { path().match( context.request()->url().path( ) ) };
+  if ( ! match.hasMatch() )
+  {
+    throw QgsServerApiNotFoundError( QStringLiteral( "Collection was not found" ) );
+  }
+  const auto collectionId { match.captured( QStringLiteral( "collectionId" ) ) };
+  // May throw if not found
+  const auto mapLayer { layerFromCollection( context, collectionId ) };
+  Q_ASSERT( mapLayer );
+
+  const auto title { mapLayer->title().isEmpty() ? mapLayer->name().toStdString() : mapLayer->title().toStdString() };
+  const auto shortName { mapLayer->shortName().isEmpty() ? mapLayer->name() : mapLayer->shortName() };
+  json _links { links( context ) };
```

Comment:
Why `_`?

Reply:
because `links()` is a function, but I can rename it of course.